src/rpdk/core/contract/resource_client.py

```diff
@@ -41,6 +41,50 @@ def prune_properties(document, paths):
     return document
 
 
+def prune_null_properties_in_path(model, paths):
+    """Prune given properties from a document.
+
+    This assumes properties will always have an object (dict) as a parent.
+    The function modifies the document in-place, but also returns the document
+    for convenience. (The return value may be ignored.)
+    """
+    document = {"properties": model.copy()}
+    for path in paths:
+        try:
+            prop, resolved_path, parent = traverse(document, path)
+        except LookupError:
+            pass  # not found means nothing to delete
+        else:
+            if not prop:
+                key = resolved_path[-1]
+                del parent[key]
+    return document["properties"]
+
+
+def prune_properties_from_model_in_path(output_model, input_model, paths):
```

Comment:
Instead of doing an in-place update (which I don't think this method is doing so), it should directly return a pruned new model. During the contract test, those model will be used for following test, in-place update will impact the model which will be used for the rest of test.

Reply:
may be I am missing something. This method is using a copy of the model. The passed in parameter (original model) should not be changed.